packages/react-i18n/src/utilities/tryAbbreviateName.ts

```diff
@@ -0,0 +1,84 @@
+import {UnicodeCharacterSet} from '../constants';
+
+import {identifyScript} from './identifyScript';
+
+// Note: A similar Ruby implementation of this function also exists at https://github.com/Shopify/shopify-i18n/blob/main/lib/shopify-i18n/name_formatter.rb.
+export function tryAbbreviateName({
+  firstName,
+  lastName,
+  idealMaxLength = 3,
+}: {
+  firstName?: string;
+  lastName?: string;
+  idealMaxLength?: number;
+}): string | undefined {
+  if (!firstName && !lastName) {
+    return undefined;
+  }
+
+  const firstNameTrimmed = firstName?.trimLeft()?.trimRight();
+  const lastNameTrimmed = lastName?.trimLeft()?.trimRight();
+
+  const combinedName = [firstNameTrimmed, lastNameTrimmed].join('');
+  if (
+    new RegExp(/\s/).test(combinedName) ||
+    new RegExp(`[${UnicodeCharacterSet.Punctuation}]`).test(combinedName)
```

Comment:
I did... doesn't work for some reason! :(

Reply:
Looking at the Regex, this might work (with the pipe)

```suggestion
    new RegExp(`[${UnicodeCharacterSet.Punctuation}|\\s]`).test(combinedName)
```